examples/restaurantbot/run.py

```diff
@@ -0,0 +1,115 @@
+import argparse
+import asyncio
+import logging
+from typing import Text, Optional
+
+import rasa.utils.io
+import rasa.train
+from examples.restaurantbot.policy import RestaurantPolicy
+from rasa.core.agent import Agent
+from rasa.core.interpreter import RasaNLUInterpreter
+from rasa.core.policies.memoization import MemoizationPolicy
+from rasa.core.policies.mapping_policy import MappingPolicy
+
+logger = logging.getLogger(__name__)
+
+
+async def parse(
+    text: Text, core_model_path: Text, nlu_model_path: Optional[Text] = None
+):
+    if nlu_model_path:
+        interpreter = RasaNLUInterpreter(nlu_model_path)
+    else:
+        interpreter = None
+
+    agent = Agent.load(core_model_path, interpreter=interpreter)
+
+    response = await agent.handle_text(text)
+
+    logger.info("Text: '{}'".format(text))
+    logger.info("Response:")
+    logger.info(response)
+
+    return response
+
+
+async def train_core(
+    domain_file: Text = "domain.yml",
+    model_path: Text = "models/core",
+    training_data_file: Text = "data/stories.md",
+):
+    agent = Agent(
+        domain_file,
+        policies=[
+            MemoizationPolicy(max_history=3),
+            MappingPolicy(),
+            RestaurantPolicy(batch_size=100, epochs=400, validation_split=0.2),
+        ],
+    )
+
+    training_data = await agent.load_data(training_data_file)
+    agent.train(training_data)
+
+    # Attention: agent.persist stores the model and all meta data into a folder.
+    # The folder itself is not zipped.
+    agent.persist(model_path)
+
+    logger.info("Model trained. Stored in '{}'.".format(model_path))
+
+    return model_path
+
+
+def train_nlu(
+    config_file="config.yml", model_path="models/nlu", training_data_file="data/nlu.md"
+):
+    from rasa.nlu.training_data import load_data
+    from rasa.nlu import config
+    from rasa.nlu.model import Trainer
+
+    training_data = load_data(training_data_file)
+    trainer = Trainer(config.load(config_file))
+    trainer.train(training_data)
+
+    # Attention: trainer.persist stores the model and all meta data into a folder.
+    # The folder itself is not zipped.
+    model_directory = trainer.persist(model_path)
+
+    logger.info("Model trained. Stored in '{}'.".format(model_path))
```

Comment:
Maybe makes more sense to say that it is stored in `model_directory` (well, the local path version of it) since this is the path you have to give to `parse`